src/main/java/seedu/command/AddCommand.java

```diff
@@ -47,10 +47,35 @@ public void execute(UserList users, Ui ui, User nowUser/*, Storage storage*/) th
                                 timeInputs[0].trim(), timeInputs[1].trim());
                         ArrayList<Event> timetable = users.getUser(i + 1).getTimetable().getTimetable(day);
                         timetable.add(newEvent);
+                        
                         try {
                             int timeStart = Integer.parseInt(timeInputs[0].trim());
                             int timeEnd = Integer.parseInt(timeInputs[1].trim());
+                            int timeStartHour = Integer.parseInt(timeInputs[0].trim().substring(0, 2));
+                            int timeStartMin = Integer.parseInt(timeInputs[0].trim().substring(2));
+                            int timeEndHour = Integer.parseInt(timeInputs[1].trim().substring(0, 2));
+                            int timeEndMin = Integer.parseInt(timeInputs[1].trim().substring(2));
+                            
+                            if (timeStartHour < 0 || timeStartHour > 23) {
+                                timetable.remove(newEvent);
+                                throw new WhereGotTimeException("Invalid Hour! It should be within "
+                                        + "24-hour format. (0000-2359)");
+                            } else if (timeStartMin < 0 || timeStartMin > 59) {
+                                timetable.remove(newEvent);
+                                throw new WhereGotTimeException("Invalid minute! It should be within "
+                                        + "24-hour format. (0000-2359)");
+                            }
 
+                            if (timeEndHour < 0 || timeEndHour > 23) {
+                                timetable.remove(newEvent);
+                                throw new WhereGotTimeException("Invalid Hour! It should be within "
+                                        + "24-hour format. (0000-2359)");
+                            } else if (timeEndMin < 0 || timeEndMin > 59) {
+                                timetable.remove(newEvent);
+                                throw new WhereGotTimeException("Invalid minute! It should be within "
+                                        + "24-hour format. (0000-2359)");
+                            }
+                            
```

Comment:
perhaps the logic can be improved by checking for time validity first, and if it's valid, proceed to add event. In other words, you shouldn't delete the event only after realising the time is not valid. This results in duplicated code.

Reply:
The logic is the same, E.g. if you check  for validity then add and you still have to take into consideration of what happens if is invalid , keep or delete. So, if you check for invalid and remove them (not adding them) , it would be the same.

Reply:
its even better to check and do action to invalidity first as after you check for validity you would still need to add and then keep track of the invalid (error message)

Reply:
If the time is not valid to begin with, it should be thrown immediately and request the user to re-enter with valid timings. i.e. the case where "check for validity then add" does not need to happen, as the program will throw an exception

Reply:
by this logic,but the invalid would still be added after the error message is shown.

Reply:
![image](https://user-images.githubusercontent.com/60384540/98150642-ef9ad300-1f09-11eb-9c40-58258f1690c1.png)
![image](https://user-images.githubusercontent.com/60384540/98150673-ff1a1c00-1f09-11eb-8042-4f4ff06523f6.png)

Reply:
I believe that implementing a checkTimeValidity with throwable in line 34 would have been suffice. The invalid times would not be added when the program throws an exception.
![image](https://user-images.githubusercontent.com/69618398/98150817-31c41480-1f0a-11eb-94dc-91e23764ec8a.png)

Reply:
Nevermind, got it to work based on your proposed logic. Will work on it. Meanwhile can merge this first? i will do another pr after this.